src/psyclone/psyir/transformations/region_trans.py

```diff
@@ -63,6 +65,43 @@ class RegionTrans(Transformation):
     # populated by sub-class.
     valid_node_types = ()
 
+    def get_node_list(self, nodes):
+        '''This is a helper function for region based transformation.
+        The parameter for any of those transformations is either a single
+        node, a schedule, or a list of nodes. This function converts this
+        into a list of nodes according to the parameter type. Note that the
+        list might be a copy of the argument or the list of children of a
+        schedule, so the result list should not be modified.
```

Comment:
This comment was based on (existing) comments in other transform where a copy of the node_list is created. The reason for this (I think)were: if you iterate over the (original) node_list, and inside of the loop you change the tree, then the node_list might change, messing up the loop. Example from ParallelRegionTrans:
```
        # Change all of the affected children so that they have
        # the region directive's Schedule as their parent. Use a slice
        # of the list of nodes so that we're looping over a local
        # copy of the list. Otherwise things get confused when
        # we remove children from the list.
        for child in node_list[:]:
            # Remove child from the parent's list of children
            node_parent.children.remove(child)
            child.parent = directive.dir_body
```
I have changed the code to _always_ return a copy of the list (even if the user provides a list of nodes ... since that list might again be from a Node in the tree). The commend was adjusted, and the code that I spotted that created a copy of the node list are not creating a copy anymore.

Reply:
OK, thanks. I was going to start worrying about transformations that do want to change the order of Nodes in a list but then I realised that they wouldn't be sub-classing RegionTrans.